### ISSUE TITLE
Linking with binary wixlib breaks incremental build

### ISSUE THREAD

**Issue by wixbot** [wixtoolset/issues#4060]
When building target which links with binary *.wixlib, the CAB from wixlib is extracted to randomly named directory under %TEMP%, which is then used as input to Link. Because new temporary file is generated every time the build occurs and this file is always newer then build output, the build of Link target is triggered.

> Originally opened by `theark` from [http://sourceforge.net/p/wix/bugs/3344/](http://sourceforge.net/p/wix/bugs/3344/)

**Comment by ScottLangham** [comment]
I think I ran into the same problem... I was going to raise this as a new bug, but will append here instead:

I've got a WiX project which is supposed to build incrementally. But, every time I build it Light.exe runs.

It seems that some tracking files are created in an intermediate folder called obj\x86\Debug. One file "MyProj.wixproj.BindContentsFileListnull.txt" contains a list of references to files that are in the Temp folder, here's one line from that file:
C:\Users\MyUser\AppData\Local\Temp\p3qht4rl\cab_2_ConditionsInstaller\0

When the solution is built, the build log (in verbose mode) says:
Building target "Link" completely.
Input file "C:\Users\MyUser\AppData\Local\Temp\fkzirn0a\cab_2_ConditionsInstaller\0" does not exist.
I believe because that file is missing, MsBuild decides that it should link again. These are temp files, so it seems sensible that they would disappear. After every build, the list of files in the BindContentsFileListnull.txt changes to point at the same files in a new sub-folder of the Temp dir.

Other wix projects I have seem fine and don't keep rebuilding.

How can I fix this one?

The version of light.exe is: 3.10.2516.0

Oh, and I've found a build flag LeaveTemporaryFiles. If I set that to true incremental building works ok, but the temporary files (about 40Mb) are not cleaned up. This isn't really acceptable though as on build servers we have problems with the disk filling up as temporary files keep appearing. We rely on a nightly script that cleans them up. If these files are needed, it seems like they should be in the intermediate folder and not the temp folder, but I don't see any option to redirect them?
